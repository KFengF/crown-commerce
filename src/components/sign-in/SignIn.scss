.sign-in {
    width: 38px;
    display: flex;
    flex-direction: column;

    .title {
        margin: 10px 0;
    }

    .buttons-container {
        display: flex;
        justify-content: space-between;
    }
}