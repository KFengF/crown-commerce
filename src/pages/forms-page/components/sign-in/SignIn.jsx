import React from 'react';
import { connect } from 'react-redux';
import FormInput from '../form-input/FormInput';
import CustomButton from '../../../../components/custom-button/CustomButton';
import { googleSignInStart, emailSignInStart } from '../../../../utils/redux/user/userActions';
import { SignInDiv, SignInTitleH2, SignInSpan, ButtonDiv } from './SignInStyles';
/* import './SignIn.scss'; */

const initialState = {
    email: '',
    password: ''
}

class SignIn extends React.Component {
    constructor(props) {
        super(props);

        this.state = initialState;
    }

    onSubmit = async event => {
        event.preventDefault();
        const { email, password } = this.state;
        const { emailSignInStart } = this.props;

        emailSignInStart(email, password);
        this.setState(initialState);
        //Limpiando los inputs

        /* antes de usar saga:
        try {
            await auth.signInWithEmailAndPassword(email, password);
            //iniciando sesion con email
            this.setState(initialState);
            //Limpiando los inputs
        } catch (error) {
            console.error('Error signing in', error);
        } */
    }

    onChange = event => {
        const { name, value } = event.target;
        this.setState({ [name]: value });
    }

    render() {
        const { email, password } = this.state;
        const { googleSignInStart } = this.props;
        return (
            <SignInDiv>
                <SignInTitleH2>I already have an account</SignInTitleH2>
                <SignInSpan>Sign in with your email and password</SignInSpan>
                <form onSubmit={ this.onSubmit } >
                    <FormInput
                        name="email"
                        value={ email }
                        type="email"
                        onChange={ this.onChange }
                        label="Email"
                        required 
                    />
                    <FormInput
                        name="password"
                        value={ password }
                        type="password"
                        onChange={ this.onChange }
                        label="Password"
                        required
                    />
                    <ButtonDiv>
                        <CustomButton type="submit" >
                            Sign in
                        </CustomButton>
                        <CustomButton
                            type="button"
                            onClick={ googleSignInStart }
                            isGoogleSignIn
                        >
                            Sign in with Google
                        </CustomButton>
                    </ButtonDiv>
                </form>
            </SignInDiv>
        )
    }
}

const mapDispatch = dispatch => ({
    googleSignInStart: () => dispatch(googleSignInStart()),
    emailSignInStart: (email, password) => 
    dispatch(emailSignInStart({ email, password }))
})

export default connect(null, mapDispatch)(SignIn);