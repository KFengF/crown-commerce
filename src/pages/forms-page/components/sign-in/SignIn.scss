.sign-in {
    width: 400px;
    display: flex;
    flex-direction: column;
    margin-right: 5%;

    .sign-in-title {
        margin-bottom: 10px;
    }

    span {
        letter-spacing: 1px;
    }

    .buttons-container {
        display: flex;
        justify-content: space-between;
    }
}