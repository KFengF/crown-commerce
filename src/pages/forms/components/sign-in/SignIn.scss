.sign-in {
    width: 400px;
    display: flex;
    flex-direction: column;
    margin-right: 5%;

    .title {
        margin: 10px 0;
    }

    .buttons-container {
        display: flex;
        justify-content: space-between;
    }
}