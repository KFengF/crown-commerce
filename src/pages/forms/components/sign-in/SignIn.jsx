import React from 'react';
import FormInput from '../form-input/FormInput';
import CustomButton from '../../../../components/custom-button/CustomButton';
import { signInWithGoogle } from '../../../../utils/firebase';
import './SignIn.scss';

class SignIn extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            email: '',
            password: ''
        }
    }

    onSubmit = event => {
        event.preventDefault();
        this.setState({ email: '', password: '' });
    }

    onChange = event => {
        const { name, value } = event.target;
        this.setState({ [name]: value });
    }

    render() {
        const { email, password } = this.state;
        return (
            <div className="sign-in" >
                <h2 className="title" >I already have an account</h2>
                <span>Sign in with your email and password</span>
                <form onSubmit={ this.onSubmit } >
                    <FormInput
                        name="email"
                        value={ email }
                        type="email"
                        onChange={ this.onChange }
                        label="Email"
                        required 
                    />
                    <FormInput
                        name="password"
                        value={ password }
                        type="password"
                        onChange={ this.onChange }
                        label="Password"
                        required
                    />
                    <div className="buttons-container" >
                        <CustomButton type="submit" >
                            Sign in
                        </CustomButton>
                        <CustomButton onClick={ signInWithGoogle } isGoogleSignIn >
                            Sign in with Google
                        </CustomButton>
                    </div>
                </form>
            </div>
        )
    }
}

export default SignIn;