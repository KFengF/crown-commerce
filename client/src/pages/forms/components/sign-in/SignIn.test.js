import React from 'react';
import { mount } from 'enzyme';
import { SignIn } from './SignIn.component';

describe('SignIn component', () => {
    let wrapper;
    let mockEmailSignInStart;
    let mockGoogleSignInStart;

    beforeEach(() => {
        mockEmailSignInStart = jest.fn();
        mockGoogleSignInStart = jest.fn();
        const mockProps = {
            emailSignInStart: mockEmailSignInStart,
            googleSignInStart: mockGoogleSignInStart
        }
        wrapper = mount(<SignIn { ...mockProps } />);
    });

    it('should render SignIn component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call googleSignInStart when clicked in the second CustomButton', () => {
        wrapper.find('CustomButton').at(1).simulate('click');
        expect(mockGoogleSignInStart).toHaveBeenCalled();
    });

    it(`should call emailSignInStart when clicked in the first CustomButton 
        and changed the inputs`, done => {
            wrapper.find('input').at(0).simulate('change', 
                { target: { name: 'email', value: 'testing@example.com' } }
            );
            wrapper.find('input').at(1).simulate('change', 
                { target: { name: 'password', value: '123456'} }
            );
            wrapper.find('form').simulate('submit', { preventDefault: () => {} });
            expect(mockEmailSignInStart).toHaveBeenCalled();
            done();
        }
    );
});