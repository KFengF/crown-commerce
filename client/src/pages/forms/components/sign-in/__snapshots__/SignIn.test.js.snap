// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render SignIn component 1`] = `ReactWrapper {}`;
